spec/ndt7-protocol.md

```diff
@@ -192,7 +192,110 @@ implementations will likely round bigger (or smaller) numbers to the nearest
 make sure that it does not mit values that a `int53` cannot represent. The
 proper action to take in this case is currently unspecified.
 
-# Reference implementation
+## Server discovery
+
+This section explains how a client can get the FQDN of one or more
+ndt7-enabled servers for the purpose of running tests. Of course, this
+section only applies to clients using M-Lab infrastructure. Clients
+using other server infrastructure MUST refer to the documentation
+provided by such infrastructure instead.
+
+Clients:
+
+1. SHOULD use [mlab-ns](https://github.com/m-lab/mlab-ns);
+
+2. MUST query for the `ndt_ssl` mlab-ns tool (see below for a
+description of how a real request would look like);
+
+3. MUST set `User-Agent` to identify themselves;
+
+4. SHOULD use the `policy=geo_options` policy that causes mlab-ns to
+return a set of nearby servers;
+
+5. if the connection to the first returned server is down, they SHOULD
+try with subsequent servers, and fail the test if all the returned
+servers appear to be down;
+
+6. MUST handle gracefully the case where M-Lab is out of capacity, which
+is identifiable by mlab-ns returning `200` along with an empty set of servers;
+
+7. MUST use the following retry policy (if applicable): extract a random
```

Comment:
retry policy is for errors. For errors we want exponential backoff. For repeated tests across time, we want a poisson process.  Which of these scenarios are you trying to give guidance for here?

Reply:
1. Yes.

2. Yes. If the test fails, I would recommend they just move on. Unless they really need a result soon, in which case exponential backoff +jitter is the right thing.  Exponential backoff is: try, wait 1 minute and retry, then wait 2 minutes and retry, then wait 4 minutes to retry, then wait 8 minutes and retry....

Reply:
Ok, tried to capture that in the text!